app/components/channel_drawer/channels_list/filtered_list/index.js

```diff
@@ -25,15 +25,62 @@ const pastDirectMessages = createSelector(
     (directChannelsFromPreferences) => directChannelsFromPreferences.filter((d) => d.value === 'false').map((d) => d.name)
 );
 
+const getTeamProfiles = createSelector(
+    getProfilesInCurrentTeam,
+    (members) => {
+        return members.reduce((memberProfiles, member) => {
+            memberProfiles[member.id] = member;
+
+            return memberProfiles;
+        }, {});
+    }
+);
+
+function getGroupDetails(currentUserId, userIdsInChannels, profiles, groupChannels) {
+    return groupChannels.reduce((groupMemberDetails, channel) => {
+        if (!userIdsInChannels.hasOwnProperty(channel.id)) {
+            return groupMemberDetails;
+        }
+
+        const members = Array.from(userIdsInChannels[channel.id]).reduce((details, member) => {
+            if (member === currentUserId) {
+                return details;
+            }
+
+            const profile = profiles[member];
+            const username = `${details.username || ''} ${profile.username}`;
```

Comment:
Also, instead of `const username = ${details.username || ''} ${profile.username}`, can you instead make the default argument to reduce include those fields so we don't need to do the null checking inside the template string? Like change the reduce function to `array.reduce(() => ..., {username: '', nickname: '', email: ''})` and make the previous line `const username = ${details.username} ${profile.username};` so it's clear what the format of the output is?